mesonbuild/interpreterbase/helpers.py

```diff
@@ -54,3 +54,19 @@ def default_resolve_key(key: mparser.BaseNode) -> str:
     if not isinstance(key, mparser.IdNode):
         raise InterpreterException('Invalid kwargs format.')
     return key.value
+
+def stringifyUserArguments(args: TYPE_var, quote: bool = False) -> str:
+    from ..coredata import UserOption
+    if isinstance(args, list):
```

Comment:
Maybe it is micro-optimisation, but I would put the `if` in order or likeliness: something like `str`, `bool`, `int`, `list`, `dict`, `UserOption`...